src/libcollections/lib.rs

```diff
@@ -135,6 +135,23 @@ mod std {
 }
 
 /// An endpoint of a range of keys.
+/// # Examples
+///
+/// ```
+/// use std::collections::BTreeMap;
+/// use std::collections::Bound::{Excluded, Included, Unbounded};
+///
+/// let mut map = BTreeMap::new();
+/// map.insert(3, "a");
+/// map.insert(5, "b");
+/// map.insert(8, "c");
+///
+/// for (key, value) in map.range((Excluded(3), Included(8))) {
```

Comment:
Yeah, that's true. I adapted this example from the one for [`BTreeMap::range`], which I agree is not ideal; as far as I could tell those kinds of methods are the only place in the (stable) standard library where `Bound` can be used right now though. I'd rather use something like the examples in [`RangeArgument`], but that's not stable yet so I wasn't sure if that is ok to use.

[`BTreeMap::range`]: https://doc.rust-lang.org/nightly/std/collections/struct.BTreeMap.html#method.range
[`RangeArgument`]: https://doc.rust-lang.org/nightly/std/collections/range/trait.RangeArgument.html#tymethod.start